credsweeper/file_handler/content_provider.py

```diff
@@ -9,6 +9,9 @@ class ContentProvider(ABC):
 
     @abstractmethod
     def __init__(self, file_path: str, change_type: Optional[str] = None, diff: Optional[List[Dict]] = None) -> None:
+        self.file_path = file_path
+        self.change_type = change_type
+        self.diff = diff
         raise NotImplementedError()
```

Comment:
But this function raises an error
Why even assign anything?

Reply:
many IDE warnings which cannot find the attributes in the base class

Reply:
> many IDE warnings which cannot find the attributes in the base class

However, adding this lines would decrease test coverage, as now this method interpreted as a real code by linters, not just as placeholder

If this constructor is never used (which it seems) we can just remove it, why keep it? 

Other option is to declare properties as functions:
https://stackoverflow.com/a/2736417